core/trino-main/src/test/java/io/trino/operator/scalar/TestTryFunction.java

```diff
@@ -54,35 +69,53 @@ public static long throwError()
     @Test
     public void testBasic()
     {
-        assertFunction(createTryExpression("42"), INTEGER, 42);
-        assertFunction(createTryExpression("DOUBLE '4.5'"), DOUBLE, 4.5);
-        assertFunction(createTryExpression("DECIMAL '4.5'"), createDecimalType(2, 1), decimal("4.5", createDecimalType(2, 1)));
-        assertFunction(createTryExpression("TRUE"), BOOLEAN, true);
-        assertFunction(createTryExpression("'hello'"), createVarcharType(5), "hello");
-        assertFunction(createTryExpression("JSON '[true, false, 12, 12.7, \"12\", null]'"), JSON, "[true,false,12,12.7,\"12\",null]");
-        assertFunction(createTryExpression("ARRAY [1, 2]"), new ArrayType(INTEGER), asList(1, 2));
-        assertFunction(createTryExpression("TIMESTAMP '2020-05-10 12:34:56.123456789'"), createTimestampType(9), timestamp(9, "2020-05-10 12:34:56.123456789"));
-        assertFunction(createTryExpression("NULL"), UNKNOWN, null);
+        assertThat(assertions.expression("try(42)"))
```

Comment:
Do you want these to be constants?

Reply:
Yes, those get constant folded anyway, so not point in trying anything else. Also, `try` is special in that the inner expression cannot be pulled out of the call without changing its semantics.